nvme.c

```diff
@@ -5317,6 +5398,629 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+	const char *crto = "CRTO=0x68 register offset";
+	const char *pmrcap = "PMRCAP=0xe00 register offset";
+	const char *pmrsts = "PMRSTS=0xe08 register offset";
+	const char *pmrebs = "PMREBS=0xe0c register offset";
+	const char *pmrswtp = "PMRSWTP=0xe10 register offset";
+	const char *pmrmscl = "PMRMSCL=0xe14 register offset";
+	const char *pmrmscu = "PMRMSCU=0xe18 register offset";
+
+	_cleanup_nvme_dev_ struct nvme_dev *dev = NULL;
+	int err;
+	enum nvme_print_flags flags;
+	bool fabrics = false;
+
+	_cleanup_nvme_root_ nvme_root_t r = NULL;
+	void *bar;
+
+	struct get_reg_config cfg = {
+		.offset = -1,
+	};
+
+	REG_ARGS(opts, cfg,
+		 OPT_FLAG("human-readable", 'H', &cfg.human_readable, human_readable),
+		 OPT_FLAG("cap",            'c', &cfg.cap,            cap),
+		 OPT_FLAG("vs",             'V', &cfg.vs,             vs),
+		 OPT_FLAG("cmbloc",         'm', &cfg.cmbloc,         cmbloc),
+		 OPT_FLAG("cmbsz",          'M', &cfg.cmbsz,          cmbsz),
+		 OPT_FLAG("bpinfo",         'b', &cfg.bpinfo,         bpinfo),
+		 OPT_FLAG("cmbsts",         'S', &cfg.cmbsts,         cmbsts),
+		 OPT_FLAG("cmbebs",         'E', &cfg.cmbebs,         cmbebs),
+		 OPT_FLAG("cmbswtp",        'W', &cfg.cmbswtp,        cmbswtp),
+		 OPT_FLAG("crto",           'r', &cfg.crto,           crto),
+		 OPT_FLAG("pmrcap",         'P', &cfg.pmrcap,         pmrcap),
+		 OPT_FLAG("pmrsts",         't', &cfg.pmrsts,         pmrsts),
+		 OPT_FLAG("pmrebs",         'e', &cfg.pmrebs,         pmrebs),
+		 OPT_FLAG("pmrswtp",        'w', &cfg.pmrswtp,        pmrswtp),
+		 OPT_FLAG("intms",          'i', &cfg.intms,          intms),
+		 OPT_FLAG("intmc",          'I', &cfg.intmc,          intmc),
+		 OPT_FLAG("cc",             'C', &cfg.cc,             cc),
+		 OPT_FLAG("csts",           'T', &cfg.csts,           csts),
+		 OPT_FLAG("nssr",           'n', &cfg.nssr,           nssr),
+		 OPT_FLAG("aqa",            'a', &cfg.aqa,            aqa),
+		 OPT_FLAG("asq",            'A', &cfg.asq,            asq),
+		 OPT_FLAG("acq",            'q', &cfg.acq,            acq),
+		 OPT_FLAG("bprsel",         'B', &cfg.bprsel,         bprsel),
+		 OPT_FLAG("bpmbl",          'p', &cfg.bpmbl,          bpmbl),
+		 OPT_FLAG("cmbmsc",         's', &cfg.cmbmsc,         cmbmsc),
+		 OPT_FLAG("nssd",           'N', &cfg.nssd,           nssd),
+		 OPT_FLAG("pmrctl",         'R', &cfg.pmrctl,         pmrctl),
+		 OPT_FLAG("pmrmscl",        'l', &cfg.pmrmscl,        pmrmscl),
+		 OPT_FLAG("pmrmscu",        'u', &cfg.pmrmscu,        pmrmscu));
+
+	err = parse_and_open(&dev, argc, argv, desc, opts);
+	if (err)
+		return err;
+
+	err = validate_output_format(output_format_val, &flags);
+	if (err < 0) {
+		nvme_show_error("Invalid output format");
+		return err;
+	}
+
+	if (cfg.human_readable)
+		flags |= VERBOSE;
+
+	r = nvme_scan(NULL);
+	bar = mmap_registers(r, dev, false);
+	if (!bar) {
+		err = get_register_property(dev_fd(dev), &bar, NVME_REG_PMRMSCU);
```

Comment:
Why the hard-coded `NVME_REG_PMRMSCU`?

Reply:
Sorry missed to implement the function `get_register_property()` so jsut fixed to change the function to `get_register_properties()` to get multiple regsiters for the offset and register flags.